app/scripts/modules/hmdaFilters.js

```diff
@@ -1,27 +1,90 @@
-            if (scope === 'hmda') {
-                scope = 'lar';
-                if (input.lineNumber === '1') {
-                    scope = 'ts';
-                }
-            }
+            var section = getFileSpecSection(scope, input.lineNumber);
             var fileSpec = HMDAEngine.getFileSpec(FileMetadata.get().activityYear);
-            var property = fileSpec[scopes[scope]][input.property];
+            var property = fileSpec[section][input.property];
             if (property !== undefined && property.hasOwnProperty('label')) {
                 return property.label;
             }
             return input.property;
         };
     }])
+    .filter('hmdaValue', ['HMDAEngine', '$filter', function(HMDAEngine, $filter) {
+
+        // Allow us to convert our string to a date
+        String.prototype.toDate = function() {
+            var year = parseInt(this.substr(0,4), 10),
+                month = parseInt(this.substr(4,2), 10) - 1,
+                day = parseInt(this.substr(6,2), 10);
+
+            if (this.length === 8) {
+                return new Date(year, month, day);
+            } else {
+                var hour = parseInt(this.substr(8,2), 10),
+                    minute = parseInt(this.substr(10,2), 10);
+                return new Date(year, month, day, hour, minute);
+            }
+        };
+
+        return function(value, scope, property) {
+            var section = getFileSpecSection(scope, value.lineNumber);
+            var fileSpec = HMDAEngine.getFileSpec(HMDAEngine.getRuleYear());
+            var propSpec = fileSpec[section][property];
+
+            if (propSpec !== undefined && propSpec.hasOwnProperty('validation')) {
+                var propVal = propSpec.validation;
+                var result;
+
+                if (propVal.type === 'percent') {
+                    return value + '%';
+                } else if (propVal.type === 'date' && propVal.match === 'yyyyMMdd') {
+                    return $filter('date')(value.toDate(), 'M/d/yyyy');
+                } else if (propVal.type === 'date' && propVal.match === 'yyyyMMddHHmm') {
+                    return $filter('date')(value.toDate(), 'M/d/yyyy H:mm');
+                } else if (propVal.type === 'currency') {
+                    result = parseInt(value) * parseInt(propVal.multiplier);
+                    return $filter('currency')(result, '$', 0);
+                }
+            }
+            return value;
+        };
+    }])
+    .filter('hmdaMacroValue', ['$filter', function($filter) {
+        String.prototype.startsWith = function(val) {
+            return this.indexOf(val) === 0;
+        };
+
+        String.prototype.endsWith = function(suffix) {
+            return this.indexOf(suffix, this.length - suffix.length) !== -1;
+        };
+
+        return function(value, key) {
+            if (key.startsWith('% of') || key.startsWith('% Difference') || key.endsWith('Percentage')) {
```

Comment:
Instead of extending a native object prototype, you could have just used the engine itself..
`HMDAEngine.starts_with(key, 'string')` and `HMDAEngine.ends_with(key, 'string')`